tests/development/content_types/test_content_type_negotiation.py

```diff
@@ -41,4 +41,7 @@ def test_application_response_type(nhsd_apim_proxy_url, accept_headers, method,
     if resp.status_code == 429:
```

Comment:
I wonder if we should move this out into a helper then we can call it with the `resp` value - that way we only have the retry logic in a single place?

I'm thinking we have 3 functions - 

* `handle_retry(resp)`
* `handle_429_retry(resp)`
* `handle_504_retry(resp)`

; where the `handle_retry` func executes the two other functions in order. The reason for this is that in certain places we want a 429 to be thrown but still want to retry on a 504, and vice versa.